### ISSUE TITLE
Changing results when including a date column

### ISSUE THREAD

**Issue by darthf1** [duckdb/duckdb#2426]
Hi!

I'm having strange results when including a date column in my result. I'm trying to create a reproducible sample, but maybe you can give me some pointers. Are there any "known" issues with date columns which could lead to different results?

How can I test if the result is the same on the current dev version of duckdb (currently running 0.3)?

### In short:
I'm running a query on multiple parquet files.  I have a query with several joins which has different results when including a date file.

When I run a specific query, where the result is grouped by `Year` and aggregated on `LineCount`, this is the result:
```
Total lines in my result: 274

Aggregated result on 274 lines:
Year	        LineCount
2018		2,217
2019		1,920
```
When I add a specific date column in the `SELECT` statement, the aggregated result suddenly changes:
```
Total lines in my result: 1359 <-- more rows makes sense, because I'm adding column to the `GROUP BY` result.

Aggregated result on 1359 lines:
Year	        LineCount
2018		1,879 <-- suddenly i have a lesser count
2019		1,919 <-- suddenly i have a lesser count
```

**Comment by Mytherin** [comment]
Should be solved in #2430. This was a problem in the Arrow fetch specifically where a number of buffers were collected to materialize the result as an Arrow table, but internally the buffers were re-used which caused them to be overwritten and caused the incorrect results to be gathered.

**Comment by darthf1** [comment]
I confirm my issue is resolved with version `0.3.1.dev285`.
